training new model, the num of training epochs is 500
the shape of batch_data: (5, 64, 48, 222)
the shape of batch_data after transformation (3, 3, 64, 48, 222)
0, the shape of x: (3, 32, 24, 222), the shape of residual: (3, 64, 48, 222)
1, the shape of x: (3, 16, 12, 222), the shape of residual: (3, 32, 24, 222)
2, the shape of x: (3, 8, 6, 222), the shape of residual: (3, 16, 12, 222)
3, the shape of x: (3, 4, 3, 128), the shape of residual: (3, 8, 6, 128)
3, the shape of x: (3, 4, 3, 128).
0, the shape of x: (3, 8, 6, 128).
1, the shape of x: (3, 16, 12, 222).
2, the shape of x: (3, 32, 24, 222).
3, the shape of x: (3, 64, 48, 222).
0, the shape of x: (3, 32, 24, 222), the shape of residual: (3, 64, 48, 222)
  0%|                              | 0/500 [00:00<?, ?it/s]
1, the shape of x: (3, 16, 12, 222), the shape of residual: (3, 32, 24, 222)
2, the shape of x: (3, 8, 6, 222), the shape of residual: (3, 16, 12, 222)
3, the shape of x: (3, 4, 3, 128), the shape of residual: (3, 8, 6, 128)
3, the shape of x: (3, 4, 3, 128).
0, the shape of x: (3, 8, 6, 128).
1, the shape of x: (3, 16, 12, 222).
2, the shape of x: (3, 32, 24, 222).
3, the shape of x: (3, 64, 48, 222).
the shape of batch_data: (5, 64, 48, 222)
  0%|                              | 0/500 [00:30<?, ?it/s]
Traceback (most recent call last):
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 369, in <module>
    trainer.train_model(Test_data, num_epochs=num_epochs)
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 310, in train_model
    err_test = self.eval_model(self.state, test_data)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/api.py", line 620, in cache_miss
    execute = dispatch._xla_call_impl_lazy(fun_, *tracers, **params)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/dispatch.py", line 236, in _xla_call_impl_lazy
    return xla_callable(fun, device, backend, name, donated_invars, keep_unused,
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/linear_util.py", line 300, in memoized_fun
    ans = call(fun, *args)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/dispatch.py", line 359, in _xla_callable_uncached
    return lower_xla_callable(fun, device, backend, name, donated_invars, False,
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/dispatch.py", line 445, in lower_xla_callable
    jaxpr, out_type, consts = pe.trace_to_jaxpr_final2(
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/interpreters/partial_eval.py", line 2081, in trace_to_jaxpr_final2
    jaxpr, out_type, consts = trace_to_subjaxpr_dynamic2(fun, main, debug_info)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/interpreters/partial_eval.py", line 2031, in trace_to_subjaxpr_dynamic2
    ans = fun.call_wrapped(*in_tracers_)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/linear_util.py", line 167, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 252, in eval_model
    u_pred = neural_solver(state, test_data[:,n_start], num_test=n_end-n_start)
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 248, in neural_solver
    u, _ = lax.fori_loop(0, num_test, forward_map, (u, state))
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/lax/control_flow/loops.py", line 1691, in fori_loop
    (_, result), _ = scan(_fori_scan_body_fun(body_fun), (lower_, init_val),
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/lax/control_flow/loops.py", line 258, in scan
    init_flat, carry_avals, carry_avals_out, init_tree, *rest = _create_jaxpr(init)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/lax/control_flow/loops.py", line 244, in _create_jaxpr
    jaxpr, consts, out_tree = _initial_style_jaxpr(
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/lax/control_flow/common.py", line 60, in _initial_style_jaxpr
    jaxpr, consts, out_tree = _initial_style_open_jaxpr(
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/lax/control_flow/common.py", line 54, in _initial_style_open_jaxpr
    jaxpr, _, consts = pe.trace_to_jaxpr_dynamic(wrapped_fun, in_avals, debug)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/interpreters/partial_eval.py", line 1985, in trace_to_jaxpr_dynamic
    jaxpr, out_avals, consts = trace_to_subjaxpr_dynamic(
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/interpreters/partial_eval.py", line 2002, in trace_to_subjaxpr_dynamic
    ans = fun.call_wrapped(*in_tracers_)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/linear_util.py", line 167, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/lax/control_flow/loops.py", line 1607, in scanned_fun
    return (i + 1, body_fun()(i, x)), None
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/api.py", line 624, in cache_miss
    top_trace.process_call(primitive, fun_, tracers, params))
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/interpreters/partial_eval.py", line 1743, in process_call
    jaxpr, out_type, consts = trace_to_subjaxpr_dynamic2(f, self.main, debug_info=dbg)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/interpreters/partial_eval.py", line 2031, in trace_to_subjaxpr_dynamic2
    ans = fun.call_wrapped(*in_tracers_)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/linear_util.py", line 167, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 241, in forward_map
    u = self.model.apply({'params': state.params, 'batch_stats': state.batch_stats}, u, self.train_hparams['dt'], train=False)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 1243, in apply
    return apply(
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/core/scope.py", line 865, in wrapper
    y = fn(root, *args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 1689, in scope_fn
    return fn(module.clone(parent=scope), *args, **kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 402, in wrapped_module_method
    return self._call_wrapped_method(fun, args, kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 705, in _call_wrapped_method
    y = fun(self, *args, **kwargs)
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 266, in __call__
    x, residual = DownsampleBlock(features=features, act_fn=self.act_fn, padding=self.padding, use_batch_norm=self.use_batch_norm,
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 402, in wrapped_module_method
    return self._call_wrapped_method(fun, args, kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 705, in _call_wrapped_method
    y = fun(self, *args, **kwargs)
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 157, in __call__
    residual = x = ConvLayer2(features=self.features, act_fn=self.act_fn, padding=self.padding, use_batch_norm=self.use_batch_norm)(x, train=train)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 402, in wrapped_module_method
    return self._call_wrapped_method(fun, args, kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 705, in _call_wrapped_method
    y = fun(self, *args, **kwargs)
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 123, in __call__
    activation = Conv_repl(features=self.features, kernel_size=(3, 3), name='conv1_repl')(x)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 402, in wrapped_module_method
    return self._call_wrapped_method(fun, args, kwargs)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/flax/linen/module.py", line 705, in _call_wrapped_method
    y = fun(self, *args, **kwargs)
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 79, in __call__
    activation = jnp.pad(x, pad_width=self.pad_width, mode=self.mode)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 1677, in pad
    pad_width = _broadcast_to_pairs(pad_width, ndim(array), "pad_width")
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 1412, in _broadcast_to_pairs
    raise ValueError(f"jnp.pad: {name} with nd={nd} has unsupported shape {nvals.shape}. "
jax._src.traceback_util.UnfilteredStackTrace: ValueError: jnp.pad: pad_width with nd=3 has unsupported shape (4, 2). Valid shapes are (3, 2), (1, 2), (2,), (1,), or ().
The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.
--------------------
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 369, in <module>
    trainer.train_model(Test_data, num_epochs=num_epochs)
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 310, in train_model
    err_test = self.eval_model(self.state, test_data)
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 252, in eval_model
    u_pred = neural_solver(state, test_data[:,n_start], num_test=n_end-n_start)
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 248, in neural_solver
    u, _ = lax.fori_loop(0, num_test, forward_map, (u, state))
  File "/work/09012/haoli1/ls6/hurricane/code/hurr_u_net_0.0.py", line 241, in forward_map
    u = self.model.apply({'params': state.params, 'batch_stats': state.batch_stats}, u, self.train_hparams['dt'], train=False)
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 266, in __call__
    x, residual = DownsampleBlock(features=features, act_fn=self.act_fn, padding=self.padding, use_batch_norm=self.use_batch_norm,
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 157, in __call__
    residual = x = ConvLayer2(features=self.features, act_fn=self.act_fn, padding=self.padding, use_batch_norm=self.use_batch_norm)(x, train=train)
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 123, in __call__
    activation = Conv_repl(features=self.features, kernel_size=(3, 3), name='conv1_repl')(x)
  File "/work/09012/haoli1/ls6/hurricane/code/U_net_hurricane.py", line 79, in __call__
    activation = jnp.pad(x, pad_width=self.pad_width, mode=self.mode)
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 1677, in pad
    pad_width = _broadcast_to_pairs(pad_width, ndim(array), "pad_width")
  File "/work/09012/haoli1/ls6/JAX/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 1412, in _broadcast_to_pairs
    raise ValueError(f"jnp.pad: {name} with nd={nd} has unsupported shape {nvals.shape}. "
ValueError: jnp.pad: pad_width with nd=3 has unsupported shape (4, 2). Valid shapes are (3, 2), (1, 2), (2,), (1,), or ().